### DISCUSSION TITLE
Custom path style integrations

### DISCUSSION THREAD

**Discussion by psi4j** [gokcehan/lf#1730] [Q&A] [answered]
I was wondering if there is a way to create custom path style integrations, for instance:
![image](https://github.com/gokcehan/lf/assets/116757358/76b7420e-4c79-45de-9998-2e2c7651d313)
Allowing for something like Powerlevel 10k or similar to generate the path seen up top in lf. 

I might be asking for a lot here, which means it may be time for me to start improving my skills in Go. Even if you guys have an idea of how I could achieve this and just want to point me in the right direction, I can start working on it instead. Thank you.

**Comment by CatsDeservePets** [comment, marked as answer]
Hello @psi4j!

`promptfmt` is the variable controlling the looks of the path bar on the top. There is no need for any `Go` here, `lf` is designed to be expandable using shell scripts. You can archive something that looks like this

<img width="307" height="34" alt="Screenshot 2025-07-16 at 3 06 11 PM" src="https://github.com/user-attachments/assets/fcbf39da-48c2-4301-8181-c24fc96a584a" />


using the following configuration inside your `lfrc` file:

```sh
# update `promptfmt` when switching directories
cmd on-cd &{{
    dir="$PWD"
    if [[ $dir == "/" ]]; then
        prefix="\033[90m/"
        rest=""
    elif [[ $dir == "$HOME"* ]]; then
        prefix="\033[37m~"
        rest="\033[90m${dir#$HOME}/"
    else
        prefix="\033[90m${dir}/"
        rest=""
    fi
    prompt=$(printf '\033[31m \033[90m  \033[0m%s%s\033[0m%%f' "$prefix" "$rest")
    # connect to `lf` using remote command
    lf -remote "send $id : set promptfmt \"$prompt\""
}}

# also update it on launch
cmd on-init :{{
    on-cd
}}
```
